### PR TITLE
Add golangci-lint, fix linting issues

### PR DESCRIPTION
**Pull request by breml**
Fixes: #62

### CODE REVIEW

File: pkg/providers/gitlab.go [hunk 1]
```diff
@@ -28,20 +28,7 @@ type gitLab struct {
...
-	score                  int
-}
-
-func (g *gitlabFileInfo) String() string {
-	if g.displayName != "" {
-		return g.displayName
-	}
-	return g.name
-}
-
```
Comment (sirlatrom) [inline pkg/providers/gitlab.go:42]: I imagine these are removed because they are unused?

File: pkg/providers/gitlab.go [hunk 2]
```diff
@@ -53,7 +40,8 @@ func (g *gitLab) Fetch() (*File, error) {
 	} else {
 		// TODO: handle case when repo doesn't have releases?
 		log.Infof("Getting latest release for %s/%s", g.owner, g.repo)
-		name, _, err := g.GetLatestVersion()
+		var name string
+		name, _, err = g.GetLatestVersion()
```
Comment (sirlatrom) [inline pkg/providers/gitlab.go:44]: I don't understand this change. Is it about re-assigning an existing variable (`err`)?

File: pkg/providers/hashicorp.go [hunk 3]
```diff
@@ -67,21 +67,11 @@ func (g *hashiCorp) listReleases(repoName string) (*hashiCorpRepo, error) {
...
 
-type hashiCorpFileInfo struct {
-	url, name string
-	score     int
-}
-
-func (g *hashiCorpFileInfo) String() string {
-	return g.name
-}
-
```
Comment (sirlatrom) [inline pkg/providers/hashicorp.go:78]: I imagine these are removed because they are unused?

File: pkg/providers/hashicorp.go [hunk 4]
```diff
@@ -90,7 +80,8 @@ func (g *hashiCorp) Fetch() (*File, error) {
 		log.Infof("Getting %s release for %s", g.tag, g.repo)
 		release, err = g.getRelease(g.repo, g.tag)
 	} else {
-		version, _, err := g.GetLatestVersion()
+		var version string
+		version, _, err = g.GetLatestVersion()
```
Comment (sirlatrom) [inline pkg/providers/hashicorp.go:84]: I don't understand the reasoning behind this change. What was the linting error that would recommend this?

**Review by sirlatrom — COMMENTED** [review]
Thanks for this! I have a few questions as I haven't run the linter yet, and I don't see what caused a few of the changes.

File: pkg/providers/gitlab.go (hunk 1 above)
Comment (breml) [reply pkg/providers/gitlab.go:42]: Yes, correct. The `unused` linter kicked in here.

File: pkg/providers/gitlab.go (hunk 2 above)
Comment (breml) [reply pkg/providers/gitlab.go:44]: With `:=` all variables on the left hand side are defined in the respective block, if not yet defined. Therefore, there is a new `err` variable defined. This `err` variable is then re-used on line 48 (`GetRelease`). The check, if the `err` is nil is done outside of the block and therefore checking against the variable defined on line 35 and not against the `err` from the `else` block. It is a combination of shadowing and ineffective assignment.

File: pkg/providers/hashicorp.go (hunk 3 above)
Comment (breml) [reply pkg/providers/hashicorp.go:78]: Yes, correct. The `unused` linter kicked in here.

File: pkg/providers/hashicorp.go (hunk 4 above)
Comment (breml) [reply pkg/providers/hashicorp.go:84]: Same problem as in gitlab provider explained above.

**Review by marcosnils — COMMENTED** [review]
Shall we add some insturctions on people how to run the linter locally before commit? Otherwise, people will not know and the CI will just complain about it without giving instructions on how to actually run the linter and with what arguments exactly.

**Comment by sirlatrom** [comment]
> 
> 
> Shall we add some insturctions on people how to run the linter locally before commit? Otherwise, people will not know and the CI will just complain about it without giving instructions on how to actually run the linter and with what arguments exactly.

@marcosnils Yes, I'd agree with that. At this stage, maybe the right thing to do would be adding a `CONTRIBUTING.md`?

**Comment by marcosnils** [comment]
hey @sirlatrom I saw you added a pre-commit config file. I guess we need to tell people to run that in their machines before commiting?

**Comment by sirlatrom** [comment]
> hey @sirlatrom I saw you added a pre-commit config file. I guess we need to tell people to run that in their machines before commiting? 

@marcosnils Right, that would make sense!

**Comment by marcosnils** [comment]
> @marcosnils Right, that would make sense!

Since we already merged the CONTRIBUITING guide let's add the instruction there

**Pull request merged**